skorch/net.py

```diff
@@ -567,63 +591,243 @@ def _apply_virtual_params(self, virtual_kwargs):
     def initialize_virtual_params(self):
         self.virtual_params_ = {}
 
-    def initialize_optimizer(self, triggered_directly=True):
+    def initialize_optimizer(self, triggered_directly=None):
         """Initialize the model optimizer. If ``self.optimizer__lr``
         is not set, use ``self.lr`` instead.
 
         Parameters
         ----------
-        triggered_directly : bool (default=True)
-          Only relevant when optimizer is re-initialized.
-          Initialization of the optimizer can be triggered directly
-          (e.g. when lr was changed) or indirectly (e.g. when the
-          module was re-initialized). If and only if the former
-          happens, the user should receive a message informing them
-          about the parameters that caused the re-initialization.
+        triggered_directly
+          Deprecated, don't use it anymore.
 
         """
-        args, kwargs = self.get_params_for_optimizer(
-            'optimizer', self.module_.named_parameters())
-
-        if self.initialized_ and self.verbose:
-            msg = self._format_reinit_msg(
-                "optimizer", kwargs, triggered_directly=triggered_directly)
-            print(msg)
+        # handle deprecated paramter
+        if triggered_directly is not None:
+            warnings.warn(
+                "The 'triggered_directly' argument to 'initialize_optimizer' is "
+                "deprecated, please don't use it anymore.", DeprecationWarning)
 
-        if 'lr' not in kwargs:
-            kwargs['lr'] = self.lr
+        named_parameters = self.get_all_learnable_params()
```

Comment:
Yes, that's the idea. As is, if a user adds `self.module2_`, it's never updated. I have the feeling that this is not expected and it could indeed be very hard to spot. If a user wanted to update `self.module2_`, they would need to override `initialize_module` and add the parameters to the existing optimizer or create a new one.

In contrast, let's assume that the user does _not_ want `self.module2_` to be updated by `self.optimizer_`. In that case, they would most likely have to touch `initialize_module` anyway. If they do, it's trivial to have `self.optimizer_` only update parameters of `self.module_`.

WDYT? Is the proposed behavior unexpected?